### ISSUE TITLE
MP2020Compability

### ISSUE THREAD

**Issue by BowenD-UCB** [materialsproject/pymatgen#2952]
**Issue**: [MP2020Compatibility](https://github.com/materialsproject/pymatgen/blob/83a87bb5039414aeb68f25162b5c1403478028d8/pymatgen/entries/compatibility.py#L821-L1097) (and all other compatibility modules) currently requires a ComputedEntry object containing VASP settings and POTCAR information to operate. This makes it inconvenient to apply corrections when only the structure and energy are available. This issue is becoming more common when dealing with large datasets (e.g., directly downloading structures from MP) and machine learning methods like M3GNet.

**Example**: M3GNet is trained on MPRelaxSet entries, but it's challenging to apply MP2020 compatibility to M3GNet predicted energies since there are no VASP settings available.

**Proposed Solution**: Add an extra key under the compatibility modules (or ComputedEntry) to simplify the process. For instance, `VASPsettings = "MPRelaxSet"` could be added. With this key, energy correction can be applied without explicitly including all VASP settings and creating a ComputedEntry.

**Comment by janosh** [comment]
@rkingsbury @shyuep I've been in that same situation and would also welcome such a change. Any reservations/things to be aware of from your side?

**Comment by shyuep** [comment]
I have no reservations on enabling this. But there are two approaches to this. One is to create a ComputedEntry.from_settings factory method. This will make it explicit that the construction is different from using the `__init__` method. Anything that uses this method will by default apply the relevant VASP POTCAR settings based on the Composition.

**Comment by rkingsbury** [comment]
I have also run into this and I agree the proposed change would be a good enhancement. I like @shyuep 's proposal. An alternate or complementary approach would be to add a kwarg to `MP2020Compatibiilty.__init__` that disables the POTCAR checks which occur at [this line](https://github.com/materialsproject/pymatgen/blob/83a87bb5039414aeb68f25162b5c1403478028d8/pymatgen/entries/compatibility.py#L944). I don't have a strong opinion as to which approach is better; perhaps the `.from_settings` method since the energy adjustments in `MP2020Compatibility` are developed for specific POTCARs.

Interestingly, it appears that you might be able to bypass the POTCAR check already by specifying something other than `vasp` in `entry.parameters.software`. 

```
if entry.parameters.get("software", "vasp") == "vasp":
            pc = PotcarCorrection(MPRelaxSet, check_hash=self.check_potcar_hash)
            pc.get_correction(entry)
```

I haven't tried this to confirm, but if I'm right this is a bug that should be fixed in any case.

**Comment by BowenD-UCB** [comment]
![image](https://user-images.githubusercontent.com/84425382/234145420-112fdf1d-6ab6-4e91-8dab-8b3a95b2ca64.png)

Seems like the bypassing is not easy

**Comment by rkingsbury** [comment]
> ![image](https://user-images.githubusercontent.com/84425382/234145420-112fdf1d-6ab6-4e91-8dab-8b3a95b2ca64.png)
> 
> Thanks @rkingsbury for the suggestion, Seems like the bypassing is not easy

I see, thanks for checking. It looks like this is actually bypassing the POTCAR check, but then failing the U values check. So in principle if you supplied the correct U values in your `parameters` and then used `software:foo` I think you could get around it, but obviously this is still a headache.

**Comment by mkhorton** [comment]
I can certainly see this being useful, although of course the checks are there for a reason -- especially, to pick one example, since MPRelaxSet itself has changed over time, it is perhaps not sufficient to say "generated with MPRelaxSet" for the purposes of guaranteeing compatibility. Still, sometimes...

**Comment by arosen93** [comment]
Linking this to #2499 since they overlap.

**Comment by arosen93** [comment]
Closed in fad0576eb85af458cdc18daf4df9e861c75971ed
